src/main/java/com/example/springfirsthw/controller/BookController.java

```diff
@@ -0,0 +1,51 @@
+package com.example.springfirsthw.controller;
+
+import com.example.springfirsthw.dto.BookDto;
+import com.example.springfirsthw.dto.CreateBookRequestDto;
+import com.example.springfirsthw.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/books")
+public class BookController {
+
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getById(id);
+    }
+
+    @PostMapping
+    public BookDto createBook(@RequestBody CreateBookRequestDto bookRequestDto) {
```

Comment:
Please don't forget to use @Valid annotation

Reply:
Please, add this annotation